commitserver/commit/hydratorhelper.go

```diff
@@ -135,20 +130,26 @@ func writeManifests(dirPath string, manifests []*apiclient.HydratedManifestDetai
 		buf := bytes.Buffer{}
 		enc := yaml.NewEncoder(&buf)
 		enc.SetIndent(2)
+
+		defer func() {
+			if cerr := enc.Close(); cerr != nil {
+				log.WithError(cerr).Error("failed to close yaml encoder")
+			}
+		}()
+
```

Comment:
I wonder if we could skip the bytes buffer entirely and encode directly to the file... not sure how that works with multiple documents.

I assume we'd then use a shared encoder instead of creating a new one for each manifest.